### ISSUE TITLE
Issue when starting docker-compose.yml services - missing data for "metadata.tables" 

### ISSUE THREAD

**Issue by mskyttner** [calpaterson/csvbase#27]
When trying out csvbase - a fantastic tool by the way, thanks! - I am struggling to start the `docker-compose.yml` services or cannot figure out how to provide the necessary data in the beginning to get the services to start.

I used the `docker-compose.yml` after successfully building the container image for csvbase locally from the Dockerfile. The command `docker-compose up` showed services in good health, but the log indicates data missing in the database.

I expected successful application startup with a webpage responding at "localhost:6001".

But in fact the log showed `psycopg2.errors.UndefinedTable: relation "metadata.tables" does not exist` on the first GET request.

## Additional details

I might not have understood how to seed the database with the necessary data for the application to start up successfully. Thankful for any hints or pointers!

```
$ docker-compose logs -f csvbase
Attaching to csvbase_csvbase_1
csvbase_1   | [2023-01-17 18:36:46 +0000] [7] [INFO] Starting gunicorn 20.1.0
csvbase_1   | [2023-01-17 18:36:46 +0000] [7] [INFO] Listening at: http://0.0.0.0:6001 (7)
csvbase_1   | [2023-01-17 18:36:46 +0000] [7] [INFO] Using worker: sync
csvbase_1   | [2023-01-17 18:36:46 +0000] [10] [INFO] Booting worker with pid: 10
csvbase_1   | 2023-01-17 18:36:47,386 INFO     csvbase.sentry                      - sentry not initialised - dsn not set
csvbase_1   | 'FLASK_ENV' is deprecated and will not be used in Flask 2.3. Use 'FLASK_DEBUG' instead.
csvbase_1   | 2023-01-17 18:36:47,391 WARNING  csvbase.web                         - CSVBASE_SECERT_KEY not set, using a random secret
csvbase_1   | 2023-01-17 18:37:34,212 ERROR    csvbase.web                         - Exception on / [GET]
csvbase_1   | Traceback (most recent call last):
csvbase_1   |   File "/usr/local/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1901, in _execute_context
csvbase_1   |     cursor, statement, parameters, context
csvbase_1   |   File "/usr/local/lib/python3.7/site-packages/sqlalchemy/engine/default.py", line 736, in do_execute
csvbase_1   |     cursor.execute(statement, parameters)
csvbase_1   | psycopg2.errors.UndefinedTable: relation "metadata.tables" does not exist
```

**Comment by calpaterson** [comment]
Thanks for taking the time to file this bug - aha, I'd actually completely forgotten these files were present.  

I don't use docker for csvbase at the moment but I will get them updated so that this way of building it works.  It would be nice if people could just use docker to try it out locally.  They aren't far away, they just aren't applying the migrations before starting up.

**Comment by mskyttner** [comment]
Aha, I see... I tried to use flake-migrate from the shell but wasn't able to get it to work - not sure what the command would be and where the necessary files are inside the container after the python package has been installed. 

It would be nice if the app when started for the first time (often the case with containers as you know), it would populate the database schema automatically.

**Comment by mskyttner** [comment]
I got it running and it is a handy and nifty tool you have created!

Nice especially for a local development setup when docker-compose does the build from the sources. 

I noticed that it looks like the "alembic upgrade head" command exits with success even if it has been run previously, so I played around with some changes where I tried putting this into the default CMD before starting the app, and that seems to work too.

If the last build stage of the Dockerfile were to include those alembic/migrations files (those are not too huge, are they?), the container dedicated to just running the migrations wouldn't be strictly needed and that would perhaps simplify a bit when using the images as provided from non-local registries. I tried adding a `docker-compose.yml.prod` file for that use case. 

[This branch has those changes](https://github.com/mskyttner/csvbase/commits/add-static-files) and when quickly trying it out I saw that the container image did not contain some of the stylesheets. Then I found the Makefile targets for putting those in the static directory and that those files were listed in the .gitignore. 

I just added them anyway, and that could perhaps be achieved in a neater way... Perhaps the makefile targets could be executed in the build steps of the Dockerfile (should work if make is present inside the build environment).
